index.php

```diff
@@ -12,24 +12,54 @@
 
 // Get XML source through the 'xml' parameter
 if (!empty($_GET['xml']) && isset($_GET['xml'])) {
-    $xml = simplexml_load_file($_GET['xml']);
+    $xmlQueryString = $_GET['xml'];
+
+    // If query is a file location over http, ensure it's https
```

Comment:
Is there a reason why you want to enforce it to use HTTPS? Some URLs don't have HTTPS and can only load over `http://`. In order to accept both HTTP and HTTPS, it would be better to perform a simple check that it's a valid URL using [`filter_var()`](https://www.w3schools.com/PHP/filter_validate_url.asp). This way it allows the API to determine if it needs to convert the URL to JSON or the XML string:

```php
$xmlQueryString = "http://example.com";

// Check if XML parameter is a URL
if (filter_var($xmlQueryString, FILTER_VALIDATE_URL)) {
    echo("$xmlQueryString is a valid URL");
    // Convert XML from URL with xmlToArray() function
} else {
    echo("$xmlQueryString is not a valid URL. Assume value is string.");
    // Convert XML from string
}
```

Reply:
Apologies, thought it was part of the use case, will remove it now

Reply:
https://github.com/factmaven/xml-to-json/pull/2/commits/0f32b995d0e930740f5e977199df230602d6640e

Reply:
Just wanted to make sure I didn't see something that you did. I figure we just need to implement a simple URL validator so we know if it's a string or not.

Reply:
Yeah, well i did remove the error for when it's `http`, but kept the check `if it contains http or https then try load the file `